cosmo_tester/framework/examples/__init__.py

```diff
@@ -15,13 +15,13 @@
 
 import json
 import os
-import uuid
 from abc import ABCMeta
 
 import pytest
 import testtools
 
 from cosmo_tester.framework import git_helper
+from cosmo_tester.framework.util import set_client_tenant
```

Comment:
Not sure if this make sense, but if you're doing a bunch of 
```
with set_client_tenant(self.manager, self.tenant)
```
Wouldn't it make sense to have this function be this class' method? Or maybe have a method that calls this function, just so you won't have to call `self.manager, self.tenant` each time.

Reply:
The problem is that it's used across a bunch of different tests and other components- e.g. it's used in the examples (which aren't parented off any common ancestor to the tests, thankfully), in the hostpool test, in the snapshots tests and image based tests.

The snapshot tests in particular don't use the class properties (and neither do some of the others), so I think trying to bake this into a common ancestor to everything would actually lead to some fairly unpleasant code.

Reply:
Yeah, fair enough